libp2p/muxers/mplex/mplex.nim

```diff
@@ -76,10 +77,11 @@ proc newStreamInternal*(m: Mplex,
     "channel slot already taken!")
 
   m.getChannelList(initiator)[id] = result
-  libp2p_mplex_channels.set(
-    m.getChannelList(initiator).len.int64,
-    labelValues = [$initiator,
-                   $m.connection.peerInfo])
+  when defined(libp2p_expensive_metrics):
```

Comment:
can't we put this behind a template in `libp2p/metrics.nim` which imports and reexports upstream metrics?

Reply:
Why obfuscate what is happening?

Reply:
because metrics are not supposed to break the flow of reading the actual logic of the code in question - they are a monitoring and debugging help, not a feature in and of their own - the less space they take up, the better - and repeating when defined everywhere breaks the flow

Reply:
> repeating when defined everywhere breaks the flow

Not by much. It's just one extra line and you know what it does from its first couple of words. Compare that with `declareExpensiveLibp2pGauge ...`

Reply:
template have autocompletion, "when defined(...)" do not forcing to check everywhere that we are correctly spelling them.

Also that makes it easy to add new things like a `{.gcsafe.}` in a single location.